crates/lib3h/src/engine/ghost_engine_wrapper.rs

```diff
@@ -68,32 +68,32 @@ where
         }
     }
 
-    fn make_callback() -> GhostCallback<
-        LegacyLib3h<Engine, EngineError>,
-        ClientRequestContext,
-        ClientToLib3hResponse,
-        EngineError,
-    > {
+    fn make_callback(
+        request_id: String,
+    ) -> GhostCallback<LegacyLib3h<Engine, EngineError>, ClientToLib3hResponse, EngineError> {
```

Comment:
Is it ok to pass request_id in as a param like this? This is the only data needed from the ClientRequestContext, so we need this here to move it into the callback closure now that context is removed

Reply:
I'm ok with this :+1: